File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s= s.toLowerCase();
        s= s.replaceAll("[^a-zA-Z0-9]", "");
        
        int ptr1=0;
        int ptr2=s.length()-1;
        while(ptr1<ptr2)
        {
            if(s.charAt(ptr1)!=s.charAt(ptr2))
            {
               
               return false;
            }
            ptr1++;
            ptr2--;
                
        }
        
        return true;
        }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        newS =""
        for ch in s:
            if ch.isalnum():
                newS = newS+ch.lower();
        l =0;
        r= len(newS)-1;
        while l<=r:
            if newS[l]!=newS[r]:
                return False;
            l=l+1;
            r=r-1;
        return True
        print(s)


        

            
        
            
        
        
            